frontend/src/lib/components/MatchPropertyFilters/MatchPropertyFilters.tsx

```diff
@@ -0,0 +1,207 @@
+import React, { CSSProperties, useEffect } from 'react'
+import { useValues, BindLogic, useActions } from 'kea'
+import { propertyFilterLogic } from '../PropertyFilters/propertyFilterLogic'
+import '../../../scenes/actions/Actions.scss'
+import { TooltipPlacement } from 'antd/lib/tooltip'
+import { AndOrPropertyFilter, AndOrPropertyGroup, AnyPropertyFilter, PropertyFilter } from '~/types'
+import { TaxonomicFilterGroupType } from 'lib/components/TaxonomicFilter/types'
+import { Placement } from '@popperjs/core'
+import { TaxonomicPropertyFilter } from 'lib/components/PropertyFilters/components/TaxonomicPropertyFilter'
+import { FilterRow } from '../PropertyFilters/components/FilterRow'
+import { Button, Col, Row, Select } from 'antd'
+import { PlusOutlined, DeleteOutlined } from '@ant-design/icons'
+import './MatchPropertyFilters.scss'
+
+interface MatchPropertyFiltersProps {
+    endpoint?: string | null
+    propertyFilters?: AndOrPropertyFilter | null | PropertyFilter[]
+    onChange: (filters: AndOrPropertyFilter) => void
+    pageKey: string
+    showConditionBadge?: boolean
+    disablePopover?: boolean
+    popoverPlacement?: TooltipPlacement | null
+    taxonomicPopoverPlacement?: Placement
+    style?: CSSProperties
+    taxonomicGroupTypes?: TaxonomicFilterGroupType[]
+    showNestedArrow?: boolean
+    eventNames?: string[]
+}
+
+export function MatchPropertyFilters({
+    propertyFilters = null,
+    onChange,
+    pageKey,
+    showConditionBadge = false,
+    popoverPlacement = null,
+    taxonomicPopoverPlacement = undefined,
+    taxonomicGroupTypes,
+    style = {},
+    showNestedArrow = false,
+    eventNames = [],
+}: MatchPropertyFiltersProps): JSX.Element {
+    const logicProps = { propertyFilters, onChange, pageKey }
+    const { filtersWithNew } = useValues(propertyFilterLogic(logicProps))
```

Comment:
@Twixes @mariusandra I need some help with this typing error : ( 

The TLDR of the typing issues in this PR is coming from me trying to pass in two different types of the original `PropertyFilter[]` and the new and/or property_groups `AndOrPropertyFilter` to `filters` and props in `propertyFilterLogic`. 



<img width="1240" alt="Screen Shot 2022-02-15 at 5 36 28 PM" src="https://user-images.githubusercontent.com/25164963/154161423-ae4319c6-1912-409f-b5b8-584c96813e8e.png">

Reply:
In the propertyFilterLogic reducer:

**The types of 'filters' are incompatible between these types.
        Type '[AndOrPropertyFilter | AnyPropertyFilter[], { setFilter: (state: AndOrPropertyFilter | AnyPropertyFilter[], { propertyGroupIndex, propertyIndex, index, ...property }: { ...; }) => AnyPropertyFilter[] | ... 1 more ... | undefined; ... 4 more ...; removeFilterGroup: (state: AndOrPropertyFilter, { filterGroup }: { ...;...' is not assignable to type 'ReducerActions<propertyFilterLogicType, AndOrPropertyFilter | AnyPropertyFilter[]> | ... 4 more ... | undefined'.**